src/status_im2/subs/ens.cljs

```diff
@@ -83,12 +77,18 @@
 
 (re-frame/reg-sub
  :ens.main/screen
- :<- [:multiaccount/usernames]
+ :<- [:ens/names]
  :<- [:profile/profile]
  :<- [:ens/preferred-name]
  :<- [:ens/registrations]
  (fn [[names multiaccount preferred-name registrations]]
-   {:names           names
-    :profile/profile multiaccount
-    :preferred-name  preferred-name
-    :registrations   registrations}))
+   (let [names             (keys names)
+         in-progress-names (reduce (fn [acc {:keys [username custom-domain?]}]
+                                     (let [full-name (ens/fullname custom-domain? username)]
+                                       (conj acc full-name)))
+                                   #{}
+                                   (vals registrations))]
+     {:names           (remove #(contains? in-progress-names %) names)
```

Comment:
i'm not sure why `in-progress-names` is needed, why not just get names in reduce?

Reply:
because `:ens/names` contains names that corresponding transaction may be still pending state, but transaction of names returned from `:ens.main/screen` need to be completed.